src/EditorFeatures/VisualBasic/Highlighting/KeywordHighlightingHelpers.vb

```diff
@@ -8,127 +8,112 @@ Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+                        If .BlockKeyword.Kind = blockKind Then
+                            Yield .Span
+                        End If
+                    End With
+
+                ElseIf checkReturns AndAlso TypeOf node Is ReturnStatementSyntax Then
+                    With DirectCast(node, ReturnStatementSyntax)
+                        Yield .ReturnKeyword.Span
+                    End With
+
+                Else
+                    Dim children = node.ChildNodes.Where(Function(child) TypeOf child IsNot T AndAlso TypeOf child IsNot LambdaExpressionSyntax)
+                    nodes.AddRangeAtHead(children)
+                End If
+            End While
+        End Function
 
         <Extension()>
         Friend Function GetRelatedStatementHighlights(Of T As SyntaxNode)(
-            node As T,
-            blockKind As SyntaxKind,
-            Optional checkReturns As Boolean = False) As IEnumerable(Of TextSpan)
+                                                                           node As T,
+                                                                           blockKind As SyntaxKind,
+                                                                  Optional checkReturns As Boolean = False
+                                                                         ) As IEnumerable(Of TextSpan)
 
-            Dim highlights As New List(Of TextSpan)
-            HighlightRelatedStatements(Of T)(node, highlights, blockKind, checkReturns)
-            Return highlights
+            Return HighlightRelatedStatements(Of T)(node, blockKind, checkReturns)
         End Function
 
         <Extension()>
         Friend Function IsIncorrectContinueStatement(node As SyntaxNode, expectedKind As SyntaxKind) As Boolean
             Dim continueStatement = TryCast(node, ContinueStatementSyntax)
-            If continueStatement IsNot Nothing Then
-                Return continueStatement.Kind <> expectedKind
-            End If
-
-            Return False
+            Return (continueStatement IsNot Nothing) AndAlso (continueStatement.Kind <> expectedKind)
         End Function
 
         <Extension()>
         Friend Function IsIncorrectExitStatement(node As SyntaxNode, expectedKind As SyntaxKind) As Boolean
             Dim exitStatement = TryCast(node, ExitStatementSyntax)
-            If exitStatement IsNot Nothing Then
-                Return exitStatement.Kind <> expectedKind
-            End If
-
-            Return False
+            Return (exitStatement IsNot Nothing) AndAlso (exitStatement.Kind <> expectedKind)
         End Function
 
         <Extension>
-        Friend Sub HighlightRelatedAwaits(node As SyntaxNode, highlights As List(Of TextSpan), cancellationToken As CancellationToken)
-            If TypeOf node Is AwaitExpressionSyntax Then
-                With DirectCast(node, AwaitExpressionSyntax)
-                    ' If there is already a highlight for the previous token and it is on the same line,
```

Comment:
Maybe able to replicate this via an extension method, to preserve existing highlights,

Reply:
@jasonmalinowski Are you ok with the divergence from merging textspans of highlights, to separated highlights.